pkg/server/sandbox_stop.go

```diff
@@ -77,16 +78,58 @@ func (c *criContainerdService) StopPodSandbox(ctx context.Context, r *runtime.St
 	glog.V(2).Infof("TearDown network for sandbox %q successfully", id)
 
 	sandboxRoot := getSandboxRootDir(c.rootDir, id)
-	if err = c.unmountSandboxFiles(sandboxRoot, sandbox.Config); err != nil {
+	if err := c.unmountSandboxFiles(sandboxRoot, sandbox.Config); err != nil {
 		return nil, fmt.Errorf("failed to unmount sandbox files in %q: %v", sandboxRoot, err)
 	}
 
-	// TODO(random-liu): [P1] Handle sandbox container graceful deletion.
+	if err := c.stopSandboxContainer(ctx, id); err != nil {
```

Comment:
Actually, we do kill and delete now. But yeah, may still need graceful kill in the future. Will add the TODO back.